utility/aws.py

```diff
@@ -0,0 +1,148 @@
+        * id: id of instance
+        * avz: Availability Zone
+        * name: The value of Tag Name if define otherwise None
+
+        Args:
+            pattern (str): Pattern of tag name like:
+                pbalogh-testing-cluster-55jx2-worker*
+
+        Returns:
+            list: contains dictionaries with instance details mentioned above
+        """
+        instances_response = self.ec2_client.describe_instances(
+            Filters=[
+                {
+                    'Name': 'tag:Name',
+                    'Values': [pattern]
+                }
+            ]
+        )['Reservations']
+        instances = []
+        for instance in instances_response:
+            instance = instance['Instances'][0]
+            id = instance['InstanceId']
+            avz = instance['Placement']['AvailabilityZone']
+            name = None
+            for tag in instance['Tags']:
+                if tag['Key'] == 'Name':
+                    name = tag['Value']
+                    break
+            instance_data = dict(
+                id=id,
+                avz=avz,
+                name=name,
+            )
+            instances.append(instance_data)
+        logger.debug("All found instances: %s", instances)
+        return instances
+
+    def create_volume_and_attach(
+        self,
+        availability_zone,
+        instance_id,
+        name,
+        device='/dev/sdx',
+        encrypted=False,
+        size=100,
+        timeout=20,
+        volume_type='gp2',
+    ):
+        """ Create volume and attach to instance
+
+        Args:
+            availability_zone (str): availability zone e.g.: us-west-1b
+            instance_id (str): id of instance where to attach the volume
+            name (str): name of volume
+            device (str): name of device where to attach (default: /dev/sdx)
+            encrypted (boolean): True if encrypted False otherwise
+                (default: False)
+            size (int): size in GB (default: 100)
+            timeout (int): timeout in seconds for volume creation (default: 20)
+            volume_type (str): 'standard'|'io1'|'gp2'|'sc1'|'st1'
+                (default: gp2)
+        """
+        volume_response = self.ec2_client.create_volume(
+            AvailabilityZone=availability_zone,
+            Encrypted=encrypted,
+            Size=size,
+            VolumeType=volume_type,
+            TagSpecifications=[
+                {
+                    'ResourceType': 'volume',
+                    'Tags': [
+                        {
+                            'Key': 'Name',
+                            'Value': name,
+                        },
+                    ]
+                },
+            ]
+        )
+        logger.debug("Response of volume creation: %s", volume_response)
+        volume = self.ec2_resource.Volume(volume_response['VolumeId'])
+        for x in range(timeout):
+            volume.reload()
```

Comment:
Can we print the volume state once here?

Reply:
we can print one more debug log here, I can add in next commit ;)

Reply:
added in commit: https://github.com/red-hat-storage/ocs-ci/pull/23/commits/6659aa4f0c41c1972ffa9f7306ae8b7a8f4f52c1